java-checks/src/main/java/org/sonar/java/checks/SwitchDefaultLastCaseCheck.java

```diff
@@ -42,22 +41,28 @@
   public void visitNode(Tree tree) {
     SwitchStatementTree switchStatementTree = (SwitchStatementTree) tree;
     Optional<CaseLabelTree> defaultLabel = getDefaultLabel(switchStatementTree);
-    if (defaultLabel.isPresent() && !defaultIsLastCase(switchStatementTree, defaultLabel.get())) {
+    if (defaultLabel.isPresent()) {
```

Comment:
this can now be written in one single pass with `optional.ifPresent()`:

```java
getDefaultLabel((SwitchStatementTree) tree).ifPresent(defaultLabel -> reportIssue(defaultLabel, "Move this default to the end of the switch."));
```